src/ProvidedTypes.fs

```diff
@@ -45,19 +45,19 @@ module Utils =
     let USome v = uoption<'T>(true, v)
     let (|UNone|USome|) (x:uoption<'T>) = if x.HasValue then USome x.Value else UNone
 
-    module StructOption = 
-        let toObj x = match x with UNone -> null | USome x -> x
-        let ofObj x = match x with null -> UNone | x -> USome x
+    module StructOption = 
+        let toObj (x: uoption<'T>) = if x.HasValue then x.Value else null
+        let ofObj x = if isNull x then UNone else USome x
 
 
-    let tryFindMulti k map = match Map.tryFind k map with Some res -> res | None -> [| |]
+    let tryFindMulti k map = Map.tryFind k map |> Option.defaultValue [| |]
 
-    let splitNameAt (nm:string) idx =
-        if idx < 0 then failwith "splitNameAt: idx < 0";
-        let last = nm.Length - 1
-        if idx > last then failwith "splitNameAt: idx > last";
-        (nm.Substring(0, idx)), 
-        (if idx < last then nm.Substring (idx+1, last - idx) else "")
+    let splitNameAt (nm:string) idx =
+        if idx < 0 then failwith "splitNameAt: idx < 0"
+        if idx >= nm.Length then failwith "splitNameAt: idx > last"
+        let beforeIdx = nm.Substring(0, idx)
```

Comment:
I'm thinking if System.Memory dependency would be added on .Net Standard 2.0 then many of these low-level string parsing operations (which should be avoided totally) could be replaced with Spans.